src/agent/configuration_parser/CMakeLists.txt

```diff
@@ -0,0 +1,44 @@
+cmake_minimum_required(VERSION 3.18)
+
+set(CMAKE_CXX_STANDARD 20)
+set(CMAKE_BUILD_TYPE Release)
+set(CMAKE_CXX_FLAGS_DEBUG "${CMAKE_CXX_FLAGS_DEBUG} -g")
+
+project(ConfigurationParser)
+message(STATUS "******* Project ConfigurationParser *********")
+
+#########################################################################
+# Ensure that we do an out of source build
+#########################################################################
+MACRO(MACRO_ENSURE_OUT_OF_SOURCE_BUILD MSG)
+     STRING(COMPARE EQUAL "${CMAKE_SOURCE_DIR}"
+     "${CMAKE_BINARY_DIR}" insource)
+     GET_FILENAME_COMPONENT(PARENTDIR ${CMAKE_SOURCE_DIR} PATH)
+     STRING(COMPARE EQUAL "${CMAKE_SOURCE_DIR}"
+     "${PARENTDIR}" insourcesubdir)
+    IF(insource OR insourcesubdir)
+        MESSAGE(FATAL_ERROR "${MSG}")
+    ENDIF(insource OR insourcesubdir)
+ENDMACRO(MACRO_ENSURE_OUT_OF_SOURCE_BUILD)
+
+MACRO_ENSURE_OUT_OF_SOURCE_BUILD(
+    "${CMAKE_PROJECT_NAME} requires an out of source build."
+)
+
+message(STATUS "CMAKE_SOURCE_DIR: ${CMAKE_SOURCE_DIR}")
+message(STATUS "CMAKE_CURRENT_SOURCE_DIR: ${CMAKE_CURRENT_SOURCE_DIR}")
+
+#########################################################################
+# Dependencies
+#########################################################################
+
+# Enable testing
+enable_testing()
+
+include_directories(include)
+include_directories(${CMAKE_SOURCE_DIR}/configuration_parser)
```

Comment:
I don't think we should include this folder. Already from line 38 the headers for the configuration parser should be visible to the rest of the project.